### ISSUE TITLE
Introduction Generative AI

### ISSUE THREAD

**Issue by Koteeshwari** [Koteeshwari/Google-Generative-AI#1]

**Comment by Koteeshwari** [comment]
**Generative AI has become a buzzword, but what is it?**
Generative AI is a type of artificial intelligence technology that can produce various types of content, including text, imagery, audio, and synthetic data.

<img width="343" height="431" alt="Image" src="https://github.com/user-attachments/assets/67ded71e-8c3c-407e-9505-36177e692c77" />

AI is a branch of computer science that deals with the creation of intelligent agents, and are systems that can reason, learn, and act autonomously.


 **let's talk about machine learning.**
Machine learning, is a subfield of AI.

<img width="508" height="785" alt="Image" src="https://github.com/user-attachments/assets/daf623d3-9086-4d43-8485-87790562e7e9" />

It is a program or system that trains a model from input data.

The trained model can make useful predictions from new (never-before-seen) data drawn from the same one used to train the model.

This means that machine learning gives the computer the ability to learn without explicit programming.

<img width="428" height="159" alt="Image" src="https://github.com/user-attachments/assets/2d134ff9-3ae8-48cf-9a55-0fc3d957ff45" />

**Machine learning models look like**

<img width="468" height="165" alt="Image" src="https://github.com/user-attachments/assets/e59ea6c2-4c38-4ce9-8796-1d6365e1a8cc" />

Two of the most common classes of machine learning models are unsupervised and supervised ML models.

The key difference between the two is that with supervised models, we have labels.

Labeled data is data that comes with a tag, like a name, a type, or a number.

<img width="446" height="240" alt="Image" src="https://github.com/user-attachments/assets/d3c2f873-c09c-4507-953d-78acc1258592" />

This graph is an example of the sort of problem that a supervised model might try to solve.
<img width="418" height="395" alt="Image" src="https://github.com/user-attachments/assets/2e502af6-8d7d-40a6-b1ed-b7efbce3c232" />

Unsupervised problems are all about discovery, about looking at the raw data, and seeing if it naturally falls into groups.
<img width="388" height="385" alt="Image" src="https://github.com/user-attachments/assets/bc9a5914-9a9e-4606-b9a2-b0d871548e16" />

<img width="810" height="353" alt="Image" src="https://github.com/user-attachments/assets/b99978a0-79b2-4338-bcf0-64deec97684f" />

<img width="346" height="317" alt="Image" src="https://github.com/user-attachments/assets/c8eb36ea-7c2c-4190-b0ce-f6ccf959eef4" />

<img width="1204" height="961" alt="Image" src="https://github.com/user-attachments/assets/29e7a171-f371-469a-87a4-bbec9de4866d" />


Artificial neural networks are inspired by the human brain.
*Like your brain, they are made up of many interconnected nodes, or neurons, that can learn to perform tasks by processing data and making predictions.
*Deep learning models typically have many layers of neurons, which allows them to learn more complex patterns than traditional machine learning models.
*Neural networks can use both labeled and unlabeled data.
*This is called semi-supervised learning.
*In semi-supervised learning, a neural network is trained on a small amount of labeled data and a large amount of unlabeled data.

<img width="440" height="987" alt="Image" src="https://github.com/user-attachments/assets/9e3cacfb-e282-4678-88eb-a8178f9f9a90" />
Deep learning models (or machine learning models in general) can be divided into two types – generative and discriminative.

A discriminative model is a type of model that is used to classify or predict labels for data points.

Discriminative models are typically trained on a dataset of labeled data points, and they learn the relationship between the features of the data points and the labels.

Example:-

<img width="292" height="429" alt="Image" src="https://github.com/user-attachments/assets/2573af83-989b-4a2a-8823-8edb1a7f926a" />

<img width="233" height="435" alt="Image" src="https://github.com/user-attachments/assets/112a610b-d322-4d66-9302-72c019d47663" />

<img width="848" height="393" alt="Image" src="https://github.com/user-attachments/assets/0fed7f16-8d37-478f-97e4-17c713aff140" />

<img width="867" height="401" alt="Image" src="https://github.com/user-attachments/assets/2170ac29-0c8a-4f90-828b-23e5d06f026b" />

If you haven't seen this for awhile, the Y=f
(x) equation calculates the dependent output of a process given different inputs.
The “Y” stands for the model output, the “f” embodies the function used in the calculation (or model), and the “X” represents the input or inputs used for the formula.
<img width="303" height="310" alt="Image" src="https://github.com/user-attachments/assets/ad8ab97d-3954-4519-b2ec-3e8bb77a26cf" />

<img width="440" height="417" alt="Image" src="https://github.com/user-attachments/assets/2f71a27a-3fc9-4bef-b307-364152d00d56" />


**What is generative AI?**

<img width="332" height="196" alt="Image" src="https://github.com/user-attachments/assets/236a9088-c99d-4c1a-b18d-11e6aa5710a2" />

<img width="389" height="406" alt="Image" src="https://github.com/user-attachments/assets/f45e4956-9e42-4393-b86d-f79387ae3f71" />

<img width="347" height="251" alt="Image" src="https://github.com/user-attachments/assets/f025de65-af2e-4c0f-aff7-03eecd5ebde7" />

<img width="660" height="264" alt="Image" src="https://github.com/user-attachments/assets/533e4606-3a3b-4838-bba5-1911c4ad4b1d" />

<img width="420" height="423" alt="Image" src="https://github.com/user-attachments/assets/8e156d9d-1351-4b27-a997-9f349890a78d" />

<img width="451" height="408" alt="Image" src="https://github.com/user-attachments/assets/0f5140d9-b3a8-40b5-824d-7cf604ae8e49" />

<img width="423" height="422" alt="Image" src="https://github.com/user-attachments/assets/2b5feff2-f663-4b76-b6bf-4f736e403980" />

<img width="361" height="284" alt="Image" src="https://github.com/user-attachments/assets/5bfd8600-d772-4285-b959-d2aece71d030" />

<img width="438" height="392" alt="Image" src="https://github.com/user-attachments/assets/e61441dc-d4c6-4c64-97c6-d0aeb8721a5c" />


1.Text-to-Text.

<img width="169" height="302" alt="Image" src="https://github.com/user-attachments/assets/76004e28-43d7-4990-8673-3f8c33dde44a" />

Text-to-text models take a natural language input and produce text output.

These models are trained to learn the mapping between a pair of texts.

2. Text-to-image

<img width="164" height="272" alt="Image" src="https://github.com/user-attachments/assets/11d6e7f8-53d0-4b23-9a27-faa8ade13c49" />

Text-to-image models are trained on a large set of images, each captioned with a short text description.
Diffusion is one method used to achieve this.

3. Text-to-video

<img width="298" height="404" alt="Image" src="https://github.com/user-attachments/assets/9b39fe26-8c04-447e-a09e-e1f883f08111" />

There's also text-to-video and text-to-3D.

*Text-to-video models aim to generate a video representation from text input.
*The input text can be anything from a single sentence to a full script, and the output is a video that corresponds to the input text.
*Similarly, Text-to-3D models generate three-dimensional objects that correspond to a user’s text description, for use in games or other 3D worlds.

4.Test-to-task

<img width="280" height="378" alt="Image" src="https://github.com/user-attachments/assets/cab33b50-e543-4215-a0a4-d2b69d72f613" />

*Text-to-task models are trained to perform a defined task or action based on text input.
*This task can be a wide range of actions such as answering a question, performing a search, making a prediction, or taking some sort of action.
*For example, a text-to-task model could be trained to navigate a web user interface or make changes to a doc through a graphical user interface.

**Foundation Models**

<img width="855" height="332" alt="Image" src="https://github.com/user-attachments/assets/06bb43ce-a0a7-4c6b-a716-f84a2a407b7e" />

*Another model that's larger than those I mentioned is a foundation model, which is a large AI model pre-trained on a vast quantity of data "designed to be adapted” (or fine-tuned) to a wide range of downstream tasks, such as sentiment analysis, image captioning, and object recognition.
*Foundation models have the potential to revolutionize many industries, including healthcare, finance, and customer service.
*Foundation models have the potential to revolutionize many industries, including healthcare, finance, and customer service.
*They can even be used to detect fraud and provide personalized customer support.

*If you're looking for foundation models, Vertex AI offers a Model Garden that includes Foundation Models.

*The language Foundation Models include chat, text, and code.

<img width="319" height="349" alt="Image" src="https://github.com/user-attachments/assets/4d9554a2-3f87-43d8-aa91-20248915269f" />

*The Vision Foundation models includes stable diffusion, which has been shown to be effective at generating high-quality images from text descriptions.

<img width="470" height="361" alt="Image" src="https://github.com/user-attachments/assets/29ce35a6-0df9-4651-a5f8-c31d2292b734" />

<img width="893" height="436" alt="Image" src="https://github.com/user-attachments/assets/0125228e-4411-4612-99e8-a120d868ede9" />

<img width="651" height="430" alt="Image" src="https://github.com/user-attachments/assets/dab676e5-416c-4b23-8490-0e965938cb7c" />

<img width="606" height="351" alt="Image" src="https://github.com/user-attachments/assets/c974b2f9-6759-4114-af69-7c03515572bd" />
**OUTPUT:-**

<img width="602" height="361" alt="Image" src="https://github.com/user-attachments/assets/f1749932-b964-4180-8f62-3765d5bff04e" />

**So to summarize, Gemini code generation can help you: Debug your lines of source code, Explain your code to you line by line, Craft SQL queries for your database, Translate code from one language to another, Generate documentation and tutorials for source code.
I'm going to tell you about three other ways Google Cloud can help you get more out of generative AI.**

**Three other ways Google Cloud can help you get more out of generative AI.**
**Vertex AI**
Vertex AI Studio lets you quickly explore and customize generative AI models that you can leverage in your applications on Google Cloud.
Vertex AI Studio helps developers create and deploy generative AI models by providing a variety of tools and resources that make it easy to get started.

<img width="463" height="233" alt="Image" src="https://github.com/user-attachments/assets/b7398d07-45b7-404b-92ce-715a2b0cc6e4" />

**For example, there is a**
<img width="370" height="395" alt="Image" src="https://github.com/user-attachments/assets/3ece0253-eb06-4035-af34-a981b09c2712" />

<img width="246" height="365" alt="Image" src="https://github.com/user-attachments/assets/bf90c70f-bb2d-4cbe-a283-b0bdc5a1048a" />

**Gemini AI**
a multimodal AI model.
*Unlike traditional language models, it's not limited to understanding text alone.

*It can analyze images, understand the nuances of audio, and even interpret programming code.

*This allows Gemini to perform complex tasks that were previously impossible for AI.

*Due to its advanced architecture, Gemini is incredibly adaptable and scalable making it suitable for diverse applications.

*Model Garden is continuously updated to include new models. And now you know absolutely everything about Generative AI.
